dace/frontend/fortran/type_dumper.py

```diff
@@ -0,0 +1,192 @@
+asts = {find_name_of_stmt(m).lower(): m for m in walk(ast, Module_Stmt)}
+partial_modules = {}
+functions_to_rename = {}
+name_dict, rename_dict = name_and_rename_dict_creator(parse_order, dep_graph)
+for i in parse_order:
+    partial_ast.current_ast = i
+
+    partial_ast.unsupported_fortran_syntax[i] = []
+    if i in ["mtime", "ISO_C_BINDING", "iso_c_binding", "mo_cdi", "iso_fortran_env", "netcdf"]:
+        continue
+
+    # try:
+    partial_module = partial_ast.create_ast(asts[i.lower()])
+    partial_modules[partial_module.name.name] = partial_module
+    # except Exception as e:
+    #    print("Module " + i + " could not be parsed ", partial_ast.unsupported_fortran_syntax[i])
+    #    print(e, type(e))
+    # print(partial_ast.unsupported_fortran_syntax[i])
+    #    continue
+    tmp_rename = rename_dict[i]
+    for j in tmp_rename:
+        # print(j)
+        if partial_ast.symbols.get(j) is None:
+            # raise NameError("Symbol " + j + " not found in partial ast")
+            if functions_to_rename.get(i) is None:
+                functions_to_rename[i] = [j]
+            else:
+                functions_to_rename[i].append(j)
+        else:
+            partial_ast.symbols[tmp_rename[j]] = partial_ast.symbols[j]
+
+    print("Parsed successfully module: ", i, " ", partial_ast.unsupported_fortran_syntax[i])
+    # print(partial_ast.unsupported_fortran_syntax[i])
+# try:
+partial_ast.current_ast = "top level"
+
+program = partial_ast.create_ast(ast)
+program.module_declarations = ast_utils.parse_module_declarations(program)
+# except:
+
+#        print(" top level module could not be parsed ", partial_ast.unsupported_fortran_syntax["top level"])
+# print(partial_ast.unsupported_fortran_syntax["top level"])
+#        return
+
+structs_lister = ast_transforms.StructLister()
+structs_lister.visit(program)
+struct_dep_graph = nx.DiGraph()
+for i, name in zip(structs_lister.structs, structs_lister.names):
+    if name not in struct_dep_graph.nodes:
+        struct_dep_graph.add_node(name)
+    struct_deps_finder = ast_transforms.StructDependencyLister(structs_lister.names)
+    struct_deps_finder.visit(i)
+    struct_deps = struct_deps_finder.structs_used
+    # print(struct_deps)
+    for j, pointing, point_name in zip(struct_deps, struct_deps_finder.is_pointer, struct_deps_finder.pointer_names):
+        if j not in struct_dep_graph.nodes:
+            struct_dep_graph.add_node(j)
+        struct_dep_graph.add_edge(name, j, pointing=pointing, point_name=point_name)
+
+dumper = FortranTypeDumper(structs_lister.structs, partial_ast)
+fortran_code = []
+for i in [
+        "config_type", "single_level_type", "thermodynamics_type", "gas_type", "cloud_type", "aerosol_type", "flux_type"
+]:
+    fortran_code.append(dumper.generate_dump_code(i))
+output_filename = "dump_all.f90"
+fortran_code_all = "\n".join(fortran_code)
+with open(output_filename, "w") as f:
+    f.write(fortran_code_all)
+print("done")
```

Comment:
I don't quite get this file, it seems like:
- a pure debugging tool?
- Not really cleaned up at all
- very application specific
- not documented

What's going on here, and is this merged purposefully? If so, please clean up and document, otherwise remove.

Reply:
@acalotoiu Do you recall if we actually need this?

Reply:
This can be removed